src/main/java/codeu/controller/ActivityFeedServlet.java

```diff
@@ -67,25 +71,62 @@ void setUserStore(UserStore userStore) {
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response)
         throws IOException, ServletException {
-
+    	
+    	
     	List<User> allUsers = this.userStore.getAllUsers();
     	List<Conversation> allConversations = this.conversationStore.getAllConversations();
     	    			
-    	List<List<Message>> allMessages = new ArrayList<>();
+    	List<Message> allMessages = new ArrayList<>();
+    	Hashtable<Message, String> messageToConversationTitle = new Hashtable();
     	
 		for (int i = 0; i < allConversations.size(); i++) {
-			UUID conversationId = allConversations.get(i).getId();
-			allMessages.add(messageStore.getMessagesInConversation(conversationId));
+			Conversation conversation = allConversations.get(i);
+			UUID conversationId = conversation.getId();
+			List<Message> messagesToAdd = messageStore.getMessagesInConversation(conversationId);
+			allMessages.addAll(messagesToAdd);
 			
+			//stores conversation title corresponding to message in a hashtable
+			String conversationTitle = conversation.getTitle();
+			for (Message message : messagesToAdd) {
+				messageToConversationTitle.put(message, conversationTitle);
+			}
+			
+		}
+		
+		List<Object> allActivity = new ArrayList<>();
```

Comment:
`Object` is really flexible and powerful...maybe too flexible! You lose type safety by using a List<Object>, which means that you won't get any warnings if a bug gets introduced that adds something to this list incorrectly. In the JSP file, you're depending on knowing that the items in the list are either Users, Conversations, or Messages, but the code can't guarantee that when they can technically be any type. Since everything will still compile (since everything is an Object), you won't see the problem until you run the code.

This isn't so bad when the project is small, but it doesn't scale well for big systems where many different people are modifying the same codebase. Eventually, someone will make a mistake with the type, and then you have to debug the code at runtime (instead of catching it with the compiler and tests).

I have two suggestions on ways to avoid this, and I'm also happy to chat in more detail if you're interested! One would be to create a new common class (something like an ActivityFeedEntry) that has the information you need in order to render an entry in the list (like the timestamp and maybe the type of message). Then you can send a List<ActivityFeedEntry> to the JSP code and print them out in the correct format. The other would be to render the list of formatted, sorted strings directly in this action.

The other advantage of refactoring things this way is that it can simplify the JSP code. Instead of having to put the code to query (for example) for the user's name in that file, you can do that in this class. That way we can write tests for it, and the JSP file can be closer to simple HTML display and style, without as much complicated business logic.